Src/GoogleApis.Auth.DotNet4.Tests/OAuth2/GoogleCredentialTests.cs

```diff
@@ -67,7 +70,40 @@ public void FromStream_ServiceAccountCredential()
         {
             var stream = new MemoryStream(Encoding.UTF8.GetBytes(DummyServiceAccountCredentialFileContents));
             var credential = GoogleCredential.FromStream(stream);
+            Assert.IsInstanceOf(typeof(JwtServiceAccountCredential), credential.UnderlyingCredential);
             Assert.IsTrue(credential.IsCreateScopedRequired);
         }
+
+        /// <summary>
+        /// Creates service account credential from stream, obtains a JWT token
+        /// from the credential and checks the access token is well-formed.
+        /// </summary>
+        /// <returns></returns>
+        [Test]
+        public async Task FromStream_ServiceAccountCredential_GetJwtAccessToken()
+        {
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes(DummyServiceAccountCredentialFileContents));
+            var credential = GoogleCredential.FromStream(stream);
+
+            // Without adding scopes, the credential should be generating JWT scopes.
+            string accessToken = await (credential as ITokenAccess).GetAccessTokenForRequestAsync(DummyAuthUri);
```

Comment:
Why do you need the conversion? If it's only because you hide that method under the interface I would suggest that you will remove the ITokenAccess from methods when you implement them. I mean that
change:
Task<string> ITokenAccess.GetAccessTokenForRequestAsync(string authUri, CancellationToken cancellationToken)
to:
Task<string> GetAccessTokenForRequestAsync(string authUri, CancellationToken cancellationToken)

I don't remember myself hiding the interface method ever. I'll try to read more about it.
After reading a little bit about it... I'm still not convinced. Why do you prefer to do so?